numpy/core/getlimits.py

```diff
@@ -212,19 +227,21 @@ def _register_known_types():
                 if hasattr(umath, 'nextafter')  # Missing on some platforms?
                 else float64_ma.huge)
     float_dd_ma = MachArLike(ld,
-                              machep=-105,
-                              negep=-106,
-                              minexp=-1022,
-                              maxexp=1024,
-                              it=105,
-                              iexp=11,
-                              ibeta=2,
-                              irnd=5,
-                              ngrd=0,
-                              eps=exp2(ld(-105)),
-                              epsneg= exp2(ld(-106)),
-                              huge=huge_dd,
-                              tiny=exp2(ld(-1022)))
+                             machep=-105,
+                             negep=-106,
+                             minexp=-1022,
+                             maxexp=1024,
+                             it=105,
+                             iexp=11,
+                             ibeta=2,
+                             irnd=5,
+                             ngrd=0,
+                             eps=exp2(ld(-105)),
+                             epsneg= exp2(ld(-106)),
+                             huge=huge_dd,
+                             tiny=exp2(ld(-1022)),
+                             smallest_normal=exp2(ld(-1022)),
+                             smallest_subnormal=exp2(ld(-16445)))
```

Comment:
I couldn't find this back so quickly - do you have a reference for this value?

Reply:
And is it supposed to be the same for f80 and f128?

Reply:
Sure! I had to look a lot for this number, and I got to this discussion in stackoverflow about float80 long double numbers,
https://stackoverflow.com/questions/25916236/long-double-subnormals-denormals-get-truncated-to-0-woverflow
where the consensus was that it must be a number around (2-16445). I tried smaller numbers given in the discussion but they failed, so this one was the smallest that worked :)

Reply:
With the latest changes I just pushed, I'm getting the same value calculating the smallest_subnormal attribute in MachAr than the ones I put in here :D 🔥

Reply:
Nice. Okay, that with the reference looks correct to me.

Reply:
Seems we made a full round.  I still think this is wrong, because this is the value I get on my computer, for longdouble. That is 80bit extended precision.  As far as I understand it, the correct value is actually very simple, both for the smallest normal and smallest subnormal, the correct value is the same as for normal double precision.

Reply:
(I guess arguably many of these values are slightly weird for double-double. Eps, is the next value with "typical" precision, and `smallest_normal` does not have extended precision...)

Reply:
I know from double-double wikipedia page, they state that the smallest_subnormal is the same that the smallest_normal. But when we calculate the value manually as in machar, it is smaller. I'm not sure then what is the value we will want to hardcode here.

Reply:
Well, technically. I think the current `smallest_normal` value is incorrect, and should be the `np.exp2(ld(-968))` (basically one full mantissa *less* than double, `-1022+54` – I guess the mantissa is `52+1`, plus 1 since we want to have no overlap, giving +54).

I would be fine with changing the value, it was probably always was a bit incorrect. @charris what do you think?

The smallest subnormal/representable value is identical to that of double precision.  I am not sure I follow what you mean about `smallest_subnormal` and `smallest_normal` being the same.  Or what you mean by calculating it manually as in `machar`.  I am not sure that the "manual calculation" works for double-double, since double-double is a bit weird and probably not quite standard conform.  Also to actually calculate the values, you would have to be on hardware that supports double-double.

Reply:
If I follow you correctly, the `smallest_normal` value for double double is incorrect, but the `smallest_subnormal` value for double double is correct? And if so, we should change the `smallest_normal` value to `np.exp2(ld(-968))`

Reply:
No, what you put is the smallest subnormal for extended precision longdouble (which most hardware uses, but we don't care about here).

I would be happy to change the smallest normal (and tiny) value.  Although, would feel a bit better if we had a second opinion from somewhere.